scripts/eks-helm-test.sh

```diff
@@ -0,0 +1,196 @@
+fi
+
+if [ -z "$ACK_CONTROLLER_RELEASE_VERSION" ]; then
+    echo "No release version specified for ACK $AWS_SERVICE controller." 1>&2
+    echo "${USAGE}"
+    exit  1
+fi
+
+if [ -z "$IRSA_ROLE_ARN" ]; then
+    echo "No IRSA Role ARN specified to setup service account for ACK $AWS_SERVICE controller." 1>&2
+    echo "${USAGE}"
+    exit  1
+fi
+
+if [ -z "$ACK_K8S_SERVICE_ACCOUNT_NAME" ]; then
+    ACK_K8S_SERVICE_ACCOUNT_NAME="ack-${AWS_SERVICE}-controller"
+fi
+
+if [ -z "$TMP_DIR" ]; then
+    TEST_ID=$(uuidgen | cut -d'-' -f1 | tr '[:upper:]' '[:lower:]')
+    TMP_DIR=$ROOT_DIR/build/tmp-eks-test-$TEST_ID
+fi
+
+echo "To run the tests, using temporary directory: $TMP_DIR"
+mkdir -p "$TMP_DIR"
+
+KUBECONFIG_PATH="${TMP_DIR}/kubeconfig"
+daws eks update-kubeconfig --name "$EKS_CLUSTER_NAME" --kubeconfig "$KUBECONFIG_PATH"
+if [[ ! -f "$KUBECONFIG_PATH" ]]; then
+    echo "Failed to setup kubeconfig for Amazon EKS cluster: $EKS_CLUSTER_NAME" 1>&2
+    echo "${USAGE}"
+    exit 1
+fi
+
+KUBECONFIG="$KUBECONFIG_PATH"
+export KUBECONFIG
+
+echo "======================================================================================================"
+echo "To poke around your test cluster manually:"
+echo "export KUBECONFIG=$KUBECONFIG_PATH"
+echo "kubectl get pods -A"
+echo "======================================================================================================"
+
+# install helm chart
+ACK_K8S_RELEASE_NAME="ack-${AWS_SERVICE}-controller"
+REMOTE_HELM_CHARTS="$HELM_REGISTRY/$HELM_REPO:${AWS_SERVICE}-$ACK_CONTROLLER_RELEASE_VERSION"
+
+HELM_EXPERIMENTAL_OCI=1
+export HELM_EXPERIMENTAL_OCI
+
+echo "Installing helm charts: $REMOTE_HELM_CHARTS on Amazon EKS cluster: $EKS_CLUSTER_NAME"
+echo "Helm chart install dryrun output location: ${TMP_DIR}/helm_install_dryrun_output"
+
+LOCAL_HELM_CHARTS="${TMP_DIR}/charts"
+
+helm chart pull "$REMOTE_HELM_CHARTS"
+helm chart export "$REMOTE_HELM_CHARTS" --destination "$LOCAL_HELM_CHARTS"
+
+LOCAL_SERVICE_HELM_CHART="$LOCAL_HELM_CHARTS/ack-${AWS_SERVICE}-controller/"
+
+echo "Dowloaded charts at: $LOCAL_SERVICE_HELM_CHART"
+
+helm install --debug --dry-run --namespace "$ACK_K8S_NAMESPACE" "$ACK_K8S_RELEASE_NAME" "$LOCAL_SERVICE_HELM_CHART" > "${TMP_DIR}/helm_install_dryrun_output"
+
+echo "Proceeding with chart install"
+
+NAMESPACE_ON_CLUSTER=$(kubectl get namespace -o json | jq -re ".items[].metadata" | jq -r "select(.name == \"$ACK_K8S_NAMESPACE\")" | jq -re ".name")
+
+if [ -z "$NAMESPACE_ON_CLUSTER" ]; then
+    echo "Namespace: $ACK_K8S_NAMESPACE does not exist on Amazon EKS cluster: $EKS_CLUSTER_NAME. Creating it."
+    kubectl create namespace "$ACK_K8S_NAMESPACE"
+fi
+
+helm install --namespace "$ACK_K8S_NAMESPACE" "$ACK_K8S_RELEASE_NAME" "$LOCAL_SERVICE_HELM_CHART"
+
+sleep 60
+
+echo "Annotating Service Account with service Role ARN."
+kubectl annotate serviceaccount -n "$ACK_K8S_NAMESPACE" $ACK_K8S_SERVICE_ACCOUNT_NAME eks.amazonaws.com/role-arn="$IRSA_ROLE_ARN"
+
+sleep 60
```

Comment:
Why do we need this sleep? Waiting for pods to be recreated with the attached SA? Does it do that automatically, or do you need to delete the pod?